pkg/codegen/docs/gen.go

```diff
@@ -2144,6 +2233,101 @@ func (dctx *docGenContext) generateModulesFromSchemaPackage(tool string, pkg *sc
 	return modules
 }
 
+func generateConstructorSyntaxData(pkg *schema.Package, languages []string) *constructorSyntaxData {
+	loader := NewStaticSchemaLoader(pkg)
+	constructorGenerator := &constructorSyntaxGenerator{
+		indentSize:             0,
+		requiredPropertiesOnly: false,
+	}
+
+	schemaProgram := constructorGenerator.generateAll(pkg, generateAllOptions{
+		includeResources: true,
+		includeFunctions: false,
+		resourcesToSkip: []string{
+			// Skipping problematic resources
+			// which block example generation for Go
+			"aws:wafv2/ruleGroup:RuleGroup",
+			"aws:wafv2/webAcl:WebAcl",
+		},
+	})
+
+	type ProgramGenerator = func(program *pcl.Program) (files map[string][]byte, diags hcl.Diagnostics, err error)
+
+	constructorSyntax := newConstructorSyntaxData()
+	for _, lang := range languages {
+		boundProgram, err := constructorGenerator.bindProgram(loader, schemaProgram)
+		if err != nil {
+			continue
+		}
+
+		safeExtract := func(generator ProgramGenerator) (files map[string][]byte, diags hcl.Diagnostics, err error) {
+			defer func() {
+				if r := recover(); r != nil {
+					err = fmt.Errorf("panic: %v", r)
+					files = map[string][]byte{}
+				}
+			}()
+			return generator(boundProgram)
+		}
+
+		switch lang {
+		case "nodejs":
+			files, diags, err := safeExtract(nodejs.GenerateProgram)
+			if !diags.HasErrors() && err == nil {
+				program := string(files["index.ts"])
+				constructorSyntax.typescript = extractConstructorSyntaxExamples(
```

Comment:
It might be worth a comment somewhere explaining the approach and why we generate one large program with all the resources and then later extract based no comments that act as delimiters (for performance).

Reply:
added doc comment to `generateConstructorSyntaxData` that explains this ✅